docs/migration.rst

```diff
@@ -0,0 +1,368 @@
+
+.. code-block:: python
+
+    import zoneinfo
+    from datetime import datetime
+
+    NYC = zoneinfo.ZoneInfo("America/New_York")
+    datetime(2020, 1, 1, tzinfo=NYC)
+
+If you have a naïve ``datetime`` (or one you'd like to "reinterpret" as being
+in another zone), use ``replace``:
+
+.. code-block:: python
+
+    datetime(2020, 1, 1).replace(tzinfo=NYC)
+
+Ambiguous and imaginary times
+#############################
+
+Whenever a time zone's UTC offset changes (such as during a Daylight Saving
+Time transition), this creates either a gap ("imaginary" times) of times that
+have been skipped over in local time or a fold (ambiguous times), where there
+are two possible local times with the same "wall time". This is the problem
+that :pep:`495` was created to address.
+
+When using a :pep:`495`-compatible time zone, use the ``fold`` attribute to
+select the behavior you expect during these times. ``fold=0`` corresponds to
+the offset that applied *before* the transition, while ``fold=1`` corresponds
+to the offset that applies *after* the transition:
+
+.. code-block:: pycon
+
+    >>> dt = datetime(2020, 11, 1, 1, tzinfo=ZoneInfo("America/Los_Angeles"))
+    >>> print(dt)
+    2020-11-01 01:00:00-07:00
+    >>> dt.tzname()
+    'PDT'
+
+    >>> dt_enfolded = dt.replace(fold=1)
+    >>> print(dt_enfolded)
+    2020-11-01 01:00:00-08:00
+    >>> dt_enfolded.tzname()
+    'PST'
+
+Since PEP 495 was introduced in Python 3.6, the ``fold`` attribute is not
+available in earlier versions of Python. However, ``dateutil`` provides a
+backport for this feature via :func:`dateutil.tz.enfold`. If you are still
+supporting Python 2, you can use ``tz.enfold``:
+
+.. code-block:: pycon
+
+    >>> dt = datetime(2020, 11, 1, 1, tzinfo=ZoneInfo("America/Los_Angeles"))
+    >>> print(dt)
+    2020-11-01 01:00:00-07:00
+    >>> dt.tzname()
+    'PDT'
+
+    >>> from dateutil import tz
+    >>> dt = datetime(2020, 11, 1, 1, tzinfo=tz.gettz("America/Los_Angeles"))
+    >>> dt.tzname()
+    'PDT'
+
+    >>> dt_enfolded = tz.enfold(dt)
+    >>> dt_enfolded.tzname()
+    'PST'
+
+
+The ``tz.enfold`` function is also compatible with the ``zoneinfo`` module, and
+can be used unconditionally in 2/3 compatible code that uses different time
+zone providers in Python 2 and 3.
+
+Semantic differences between ``is_dst`` and ``fold``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As mentioned in the previous section, during a fold or a gap, the offset
+information that applies is ill-defined. With ``pytz`` you disambiguate between
+these choices by selecting which side of the transition you want to interpret
+your naïve datetime as with ``is_dst``.  With :pep:`495`, you choose that by
+setting the ``fold`` attribute of the ``datetime``. Unfortunately, ``is_dst``
+and ``fold`` do not cleanly map onto one another, because ``is_dst`` intends to
+choose whether to interpret the time as "DST" vs "STD", whereas ``fold``
```

Comment:
is STD the usual terminology here? I usually say DT vs ST, but I guess some people can't resist an anagram :)

Reply:
That's what `pytz` calls it:

```
>>> import pytz
>>> from datetime import *
>>> pytz.timezone("America/New_York").localize(datetime(2020, 1, 1))
datetime.datetime(2020, 1, 1, 0, 0, tzinfo=<DstTzInfo 'America/New_York' EST-1 day, 19:00:00 STD>)
```

The abbreviations (and even the concepts) are not amazingly relevant to the `fold` case so I figure it's best to stick to the `pytz` nomenclature. I have updated it to be less prominent, though (still needs to be mentioned in reference to the diagrams).